lib/pavilion/plugins/commands/cat.py

```diff
@@ -44,28 +45,25 @@ def run(self, pav_cfg, args):
                           file=sys.stderr, color=output.RED)
             return errno.EEXIST
 
-        return print_file(job_dir/args.file)
+        return self.print_file(job_dir / args.file)
 
+    def print_file(self, file):
+        """Print the file at the given path.
+        :param path file: The path to the file to print.
+        """
 
-def print_file(file):
-    """Print the file at the given path.
-    :param path file: The path to the file to print.
-    """
+        try:
+            with file.open() as file:
+                while True:
+                    block = file.read(4096)
+                    if not block:
+                        break
+                    output.fprint(block, file=self.outfile, end="")
 
-    try:
-        with file.open() as file:
-            while True:
-                block = file.read(4096)
-                if not block:
-                    break
-                output.fprint(block, file=sys.stdout)
-
-    except IsADirectoryError:
-        output.fprint("{} is a directory.".format(file), sys.stderr,
-                      color=output.RED)
-        return errno.EINVAL
-
-    except FileNotFoundError:
-        output.fprint("file '{}' does not exist.".format(file), sys.stderr,
-                      color=output.RED)
-        return errno.EEXIST
+        except (FileNotFoundError,
+                IsADirectoryError,
+                OSError,
+                PermissionError) as err:
+            output.fprint("Error reading file '{}': {}".format(file), err,
```

Comment:
:man_facepalming: - I should be more explicit. For generic errors that you don't know the cause for (OSError, IOError), give a generic message. For errors with an obvious cause, give a _specific_ mesage like you were doing before. 

More specific errors are almost always better. :)